src/libostree/ostree-repo-pull.c

```diff
@@ -2133,6 +2195,65 @@ repo_remote_fetch_summary (OstreeRepo    *self,
   return ret;
 }
 
+static void
+delta_superblock_fetch_on_complete (GObject        *object,
+                         GAsyncResult   *result,
+                         gpointer        user_data)
+{
+  OstreeFetcher *fetcher = (OstreeFetcher *)object;
+  FetchDeltaSuperBlockData *fetch_data = user_data;
+  OtPullData *pull_data = fetch_data->pull_data;
+  GError *local_error = NULL;
+  GError **error = &local_error;
+  GInputStream* input = _ostree_fetcher_stream_uri_finish (fetcher, result, error);
+  GBytes* delta_superblock_data = NULL;
+
+  if (input)
+    {
+      delta_superblock_data = g_input_stream_read_bytes ( input, OSTREE_MAX_METADATA_SIZE, pull_data->cancellable, error);
```

Comment:
Ah, I didn't notice these comments before.

>What I mean is that if this sets the error, we shouldn't go on and call delta_superblock_process. Although I see that ignores NULL...the rest of the async code uses goto out and I'd like to continue to match that style.

This is what I have in my further branch: https://github.com/Mathnerd314/ostree/blob/4f84489/src/libostree/ostree-repo-pull.c#L1645-L1655. It's more reasonable than this code.

I'm starting to think it'll be easier to just keep all development in this PR, and split out the "useful" or less risky parts into separate PR's.

> I'd like to at least land critical things like the memleak fixes and fix existing error handling soon without pulling in right away all of the pull code path reworks which are higher risk.

I split out a subset of the "free fetch data" commit, it might help: #382.